src/build/buildTask.ts

```diff
@@ -121,14 +126,36 @@ export class BuildTask {
         : vscode.TaskRevealKind.Silent;
 
     if (!cmakeCacheExists) {
-      let compilerArgs = (idfConf.readParameter(
-        "idf.cmakeCompilerArgs",
-        this.currentWorkspace
-      ) as Array<string>) || [
-        "-G=Ninja",
-        "-DPYTHON_DEPS_CHECKED=1",
-        "-DESP_PLATFORM=1",
-      ];
+      const espIdfVersion = await getEspIdfFromCMake(this.idfPathDir);
+      let defaultCompilerArgs;
+      if (espIdfVersion === "x.x") {
+        Logger.warn(
+          "Could not determine ESP-IDF version. Using default compiler arguments for the latest known version."
+        );
+        defaultCompilerArgs = [
+          "-G=Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      } else if (compareVersion(espIdfVersion, "4.4") >= 0) {
+        defaultCompilerArgs = [
+          "-G=Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      } else {
+        defaultCompilerArgs = [
+          "-G",
+          "Ninja",
+          "-DPYTHON_DEPS_CHECKED=1",
+          "-DESP_PLATFORM=1",
+        ];
+      }
+      let compilerArgs =
```

Comment:
When you read this value you will get the default value specified in package.json, so this fallback doesn't really work. 

We could either set `idf.cmakeCompilerArgs` empty by default in package.json (this would also not work with having default `[]`) or override it based on ESP-IDF version.

Reply:
You are right, thanks!
I've updated the PR by giving `idf.cmakeCompilerArgs` a default value inside package.json of `[]`, but I've made it work by changing the validation in `buildTask.ts`